### ISSUE TITLE
User property file classpath loading assumes a flat classpath

### ISSUE THREAD

**Issue by jamesnetherton** [castor-data-binding/castor#50]
The `AbstractProperties` class attempts to load a properties file with:

`URL url = getClass().getResource(filename);`

This is not guaranteed to work in modular class loading environments such as in OSGi or some JavaEE containers.

Perhaps it'd be better to use the `_applicationClassLoader` to load resources. But even this is currently not guaranteed to work at present because `XMLContext` [forces a specific ClassLoader](https://github.com/castor-data-binding/castor/blob/1642862af191f74c6edaebf374b4b755ff6e1b21/xml/src/main/java/org/exolab/castor/xml/XMLContext.java#L53), which itself may not be the correct one.

**Comment by wguttmn** [comment]
Fell free to provide a pull request for both "points", in other words one that fixes the problem in `AbstractProperties` and another one that highlights the second point with an example that can be used to (re)play the problem.

**Comment by jamesnetherton** [comment]
Thanks, I'll try to put a pull request together shortly.

Just to clarify, you want me to provide a second PR with an example that shows why the `_applicationClassLoader` may not be the correct ClassLoader, based on the one provided by `XMLContext` (before making any changes to fix this)? 

If so, what form does this take? A unit test? Or an example application like the ones in the `examples` directory.

**Comment by wguttmn** [comment]
Yes, a Unit test would be fine. In case you want to show me something in the context of a webapp, why not provide me with a Spring boot app that highlights the problem(s) at hand.
